lib/Frontend/Frontend.cpp

```diff
@@ -261,233 +262,300 @@ ModuleDecl *CompilerInstance::getMainModule() {
-      // Implicitly import the SwiftOnoneSupport module in non-optimized
-      // builds. This allows for use of popular specialized functions
-      // from the standard library, which makes the non-optimized builds
-      // execute much faster.
-      Invocation.getFrontendOptions()
-                .ImplicitImportModuleNames.push_back(SWIFT_ONONE_SUPPORT);
-    }
     break;
   }
-  }
 
   auto clangImporter =
     static_cast<ClangImporter *>(Context->getClangModuleLoader());
 
-  ModuleDecl *underlying = nullptr;
-  if (options.ImportUnderlyingModule) {
-    underlying = clangImporter->loadModule(SourceLoc(),
-                                           std::make_pair(MainModule->getName(),
-                                                          SourceLoc()));
-    if (!underlying) {
-      Diagnostics.diagnose(SourceLoc(), diag::error_underlying_module_not_found,
-                           MainModule->getName());
-    }
-  }
+  ModuleDecl *objCModuleUnderlyingMixedFramework =
+      Invocation.getFrontendOptions().ImportUnderlyingModule
+          ? importUnderlyingModule(clangImporter)
+          : nullptr;
 
-  ModuleDecl *importedHeaderModule = nullptr;
-  StringRef implicitHeaderPath = options.ImplicitObjCHeaderPath;
-  if (!implicitHeaderPath.empty()) {
-    if (!clangImporter->importBridgingHeader(implicitHeaderPath, MainModule)) {
-      importedHeaderModule = clangImporter->getImportedHeaderModule();
-      assert(importedHeaderModule);
-    }
-  }
+  ModuleDecl *importedHeaderModule = importBridgingHeader(clangImporter);
 
   SmallVector<ModuleDecl *, 4> importModules;
-  if (!options.ImplicitImportModuleNames.empty()) {
-    for (auto &ImplicitImportModuleName : options.ImplicitImportModuleNames) {
-      if (Lexer::isIdentifier(ImplicitImportModuleName)) {
-        auto moduleID = Context->getIdentifier(ImplicitImportModuleName);
-        ModuleDecl *importModule = Context->getModule(std::make_pair(moduleID,
-                                                                 SourceLoc()));
-        if (importModule) {
-          importModules.push_back(importModule);
-        } else {
-          Diagnostics.diagnose(SourceLoc(), diag::sema_no_import,
-                               ImplicitImportModuleName);
-          if (Invocation.getSearchPathOptions().SDKPath.empty() &&
-              llvm::Triple(llvm::sys::getProcessTriple()).isMacOSX()) {
-            Diagnostics.diagnose(SourceLoc(), diag::sema_no_import_no_sdk);
-            Diagnostics.diagnose(SourceLoc(),
-                                 diag::sema_no_import_no_sdk_xcrun);
-          }
-        }
-      } else {
-        Diagnostics.diagnose(SourceLoc(), diag::error_bad_module_name,
-                             ImplicitImportModuleName, false);
-      }
-    }
+  fillInModulesToImportFromImplicitImportModuleNames(importModules);
+
+  if (Invocation.getInputKind() == InputFileKind::IFK_Swift_REPL) {
+    supplyREPLFileWithImports(implicitModuleImportKind,
```

Comment:
This doesn't seem like the right name since it's actually creating the REPL file as well.

Reply:
Changed to createRELPFileWithImports